src/wagtail_personalisation/views.py

```diff
@@ -81,9 +82,12 @@ def copy_page_view(request, page_id, segment_id):
             'is_segmented': True,
         }
 
-        new_page = page.copy(update_attrs=update_attrs, copy_revisions=False)
+        try:
+            variant = Page.objects.get(slug=slug, depth=page.depth)
```

Comment:
Shouldn't we specify the parent instead. This will still given an error with the following structure:

```
   /some-page/my-page
   /other-page/my-page
```